builder/builder.go

```diff
@@ -200,54 +184,65 @@ func (b *Builder) InitMix(upstreamVer string, mixVer string, allLocal bool, allU
 		return err
 	}
 
-	// Set up mix metadata
-	// Deprecate '.clearurl' --> 'upstreamurl'
-	if _, err := os.Stat(filepath.Join(b.Config.Builder.VersionPath, ".clearurl")); err == nil {
-		b.UpstreamURLFile = ".clearurl"
-		fmt.Println("Warning: '.clearurl' has been deprecated. Please rename file to 'upstreamurl'")
-	}
-	if err := ioutil.WriteFile(filepath.Join(b.Config.Builder.VersionPath, b.UpstreamURLFile), []byte(upstreamURL), 0644); err != nil {
-		return err
-	}
-	b.UpstreamURL = upstreamURL
+	//TODO: Check before overwrite once old config is removed since new config already have default set
+	b.Config.Builder.UpstreamURL = upstreamURL
+	b.Config.Builder.UpstreamVer = upstreamVer
+	b.Config.Builder.MixVer = mixVer
 
-	if upstreamVer == "latest" {
+	if b.Config.Builder.UpstreamVer == "latest" {
 		ver, err := b.getLatestUpstreamVersion()
 		if err != nil {
 			return errors.Wrap(err, "Failed to retrieve latest published upstream version")
 		}
-		upstreamVer = ver
-	}
 
-	fmt.Printf("Initializing mix version %s from upstream version %s\n", mixVer, upstreamVer)
+		b.Config.Builder.UpstreamVer = ver
 
-	// Deprecate '.clearversion' --> 'upstreamversion'
-	if _, err := os.Stat(filepath.Join(b.Config.Builder.VersionPath, ".clearversion")); err == nil {
-		b.UpstreamVerFile = ".clearversion"
-		fmt.Println("Warning: '.clearversion' has been deprecated. Please rename file to 'upstreamversion'")
-	}
-	if err := ioutil.WriteFile(filepath.Join(b.Config.Builder.VersionPath, b.UpstreamVerFile), []byte(upstreamVer), 0644); err != nil {
-		return err
+		if config.UseNewConfig {
+			if err := b.Config.SaveConfig(b.BuildConf); err != nil {
```

Comment:
Not understanding yet why this call is inside the `if b.Config.Builder.UpstreamVer == "latest"` block. Don't you need to write back out the config either way after overwriting all the `b.Config.Builder.*` objects a few lines above?